homeassistant/components/light/aurora.py

```diff
@@ -0,0 +1,153 @@
+"""
+Support for Nanoleaf Aurora platform.
+
+Based in large parts upon Software-2's ha-aurora and fully
+reliant on Software-2's nanoleaf-aurora Python Library, see
+https://github.com/software-2/ha-aurora as well as
+https://github.com/software-2/nanoleaf
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.nanoleaf_aurora/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_COLOR_TEMP, ATTR_EFFECT, ATTR_HS_COLOR,
+    SUPPORT_EFFECT, SUPPORT_BRIGHTNESS, SUPPORT_COLOR_TEMP,
+    SUPPORT_COLOR, PLATFORM_SCHEMA, Light)
+from homeassistant.const import CONF_HOST, CONF_TOKEN, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import color as color_util
+from homeassistant.util.color import \
+    color_temperature_mired_to_kelvin as mired_to_kelvin
+
+REQUIREMENTS = ['nanoleaf==0.4.1']
+
+SUPPORT_AURORA = (SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP | SUPPORT_EFFECT |
+                  SUPPORT_COLOR)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Optional(CONF_NAME, default='Aurora'): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup Nanoleaf Aurora device."""
+    import nanoleaf
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    token = config.get(CONF_TOKEN)
+    aurora_light = nanoleaf.Aurora(host, token)
+    aurora_light.hass_name = name
```

Comment:
Why put the name on the aurora light instance? You can pass the name separately when instantiating the entity.

Reply:
That is a very good question, probably not needed. I'll have a look at it this weekend, thank you for the comments!